ceph/rados/core_workflows.py

```diff
@@ -4615,3 +4616,56 @@ def get_rados_df(self, pool_name: str = None):
         out = self.run_ceph_command(cmd=_cmd, client_exec=True)
 
         return out["pools"][0] if pool_name else out
+
+    def set_service_managed_type(self, service_type, unmanaged) -> bool:
+        """
+        Method to set the service to either managed or unmanaged
+        Args:
+            unmanaged: True or false, for the service management
+            service_type : service types are- mon,mgr,osd,rgw, mds
+        returns:
+            Pass -> True, Fail -> false
+        """
+        cmd_export = f"ceph orch ls {service_type} --export"
+        out = self.run_ceph_command(cmd=cmd_export, client_exec=True)[0]
+        if unmanaged:
+            log.debug(
+                f"Setting the {service_type} service as unmanaged by cephadm. current status : {out}"
+            )
+            out["unmanaged"] = "true"
+        else:
+            log.debug(
+                f"Setting the {service_type} service as unmanaged by cephadm. current status : {out}"
+            )
+            out["unmanaged"] = "false"
+
+        file_name = (
+            f"/tmp/{service_type}_spec_{self.set_service_managed_type.__name__}.yaml"
+        )
+        # Creating service config file
+        self.client.exec_command(sudo=True, cmd=f"touch {file_name}")
+        json_out = json.dumps(out)
+        # Adding the spec rules into the file
+        cmd = f"echo '{json_out}' > {file_name}"
+        self.client.exec_command(cmd=cmd, sudo=True)
+
+        log.debug(f"Contents of {service_type} spec file : {out}")
+        apply_cmd = f"ceph orch apply -i {file_name}"
+        log.info(f"Applying the spec file via cmd : {apply_cmd}")
+        self.client.exec_command(cmd=apply_cmd, sudo=True)
+        time.sleep(10)
+        out = self.list_orch_services(service_type="osd", export=True)[0]
```

Comment:
Service type hardcoded as OSD. make it generalized. `{service_type}`

Reply:
Changed the service_type